cli/cmd/report_rules.go

```diff
@@ -0,0 +1,409 @@
+//
+// Author:: Darren Murray(<darren.murray@lacework.net>)
+// Copyright:: Copyright 2021, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cmd
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/fatih/structs"
+	"github.com/olekukonko/tablewriter"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/lacework/go-sdk/api"
+)
+
+var (
+	// report-rules command is used to manage lacework report rules
+	reportRulesCommand = &cobra.Command{
+		Use:     "report-rule",
+		Aliases: []string{"report-rules", "rr"},
+		Short:   "Manage report rules",
+		Long: `Manage report rules to route reports to one or more email alert channels.		
+
+A report rule has four parts:
+
+  1. Email alert channel(s) that should receive the report
+  2. One or more severities to include
+  3. Resource group(s) containing the subset of your environment to consider
+  4. Notification types containing which report information to send
+`,
+	}
+
+	// list command is used to list all lacework report rules
+	reportRulesListCommand = &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Short:   "List all report rules",
+		Long:    "List all report rules configured in your Lacework account.",
+		Args:    cobra.NoArgs,
+		RunE: func(_ *cobra.Command, _ []string) error {
+			cli.StartProgress(" Fetching report rules...")
+			defer cli.StopProgress()
```

Comment:
I am not sure you can defer this function because maybe it will conflict with `OutputHuman()`

Reply:
👍  Yeh, will just have to call stopProgress twice inside the error and after success